File: ABC/201-250/230/b.cpp
#include <iostream>
using namespace std;

int main(){
  string t = "";
  string temp = "oxx";
  string s;
  cin >> s;
  for(int i=0;i<10;i++) t += temp;
  for(int i=0;i<3;i++){
    if(t.substr(i,s.size()) == s){
      cout << "Yes" << endl;
      return 0;
    }
  }
  cout << "No" << endl;
  return 0;
}

File: ABC/201-250/230/b.py
def main():
  s = input()
  t = ""
  for _ in range(10):
    t += "oxx"

  for i in range(3):
    if t[i:i+len(s)] == s:
      print("Yes")
      exit()

  print("No")

if __name__ == '__main__':
  main()


